ddl/cluster.go

```diff
@@ -140,13 +143,85 @@ func checkFlashbackCluster(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job, f
 	return nil
 }
 
+func addToSlice(schema string, tableName string, tableID int64, nonFlashbackIDs, allIDs []int64) ([]int64, []int64) {
+	if filter.IsSystemSchema(schema) && !strings.HasPrefix(tableName, "stats_") {
+		nonFlashbackIDs = append(nonFlashbackIDs, tableID)
+	}
+	allIDs = append(allIDs, tableID)
+	return nonFlashbackIDs, allIDs
+}
+
+// GetFlashbackKeyRanges get keyRanges for flashback cluster
+func GetFlashbackKeyRanges(sess sessionctx.Context, startKey kv.Key) ([]kv.KeyRange, error) {
+	schemas := sess.GetDomainInfoSchema().(infoschema.InfoSchema).AllSchemas()
+
+	var keyRanges []kv.KeyRange
+	var allPhysicalIDs, nonFlashbackPhysicalIDs []int64
+	for _, db := range schemas {
+		for _, table := range db.Tables {
+			if !table.IsBaseTable() || table.ID > meta.MaxGlobalID {
+				continue
+			}
+			nonFlashbackPhysicalIDs, allPhysicalIDs = addToSlice(db.Name.L, table.Name.L, table.ID, nonFlashbackPhysicalIDs, allPhysicalIDs)
```

Comment:
> if table.Partition != nil

For partitioned tables, (at least for now) we don't need to explicitly add its table id to physical IDs since there's no data prefixed with it.

However, this might not be the case in the future(for example, the 'global index' for partitioned tables(/cc @tiancaiamao and @mjonss ). So it might be a good idea to keep it as now.